apps/zotonic_launcher/bin/zotonic

```diff
@@ -18,12 +18,42 @@ cd -- "${ZOTONIC}" || \exit 1
 
 case "$1" in
     completion)
-        COMMAND_NAMES="$(${ZOTONIC_BIN}/zotonic.escript command_names)"
-        # TODO: Check OS (this works in Ubuntu 21.10)
-        OUTPUT="$ZOTONIC_BIN/zotonic-completion.bash"
-        # TODO: Completion for options
-        printf "#/usr/bin/env bash\n\n_zotonic_completions()\n{\n\tif [ \"\${#COMP_WORDS[@]}\" != \"2\" ]; then\n\t\treturn\n\tfi\n\n\tCOMPREPLY=(\$(compgen -W %s \"\${COMP_WORDS[1]}\"))\n}\n\ncomplete -F _zotonic_completions zotonic" "$COMMAND_NAMES" >| "$OUTPUT"
-        bash -c "source $OUTPUT"
+        unameOut="$(uname -s)"
+        case "${unameOut}" in
+            Linux*)     machine=Linux;;
+            Darwin*)    machine=Mac;;
+            CYGWIN*)    machine=Cygwin;;
+            MINGW*)     machine=MinGw;;
+            *)          machine="UNKNOWN:${unameOut}"
```

Comment:
Maybe also need add: `SunOS`, `GNU`, `FreeBSD`, `NetBSD`, `OpenBSD`, `DragonFly`, `Msys`. All those platforms can use `shell` too.

Reply:
Thanks for catching this @[vkatsuba](https://github.com/vkatsuba). Currently using on several FreeBSD machines, and possibly DragonFly in the near future.
> ❯ uname -s
FreeBSD

> #uname -s
DragonFly